lib/core/include/irods/rcMisc.h

```diff
@@ -360,9 +360,15 @@ int getAttriInAttriArray(const char* objPath,
                          int* outDataMode,
                          char** outChksum);
 
+// clang-format off
+__attribute__((deprecated("trimSpaces is deprecated. Use boost's trim instead.")))
 char* trimSpaces(char* str);
+// clang-format on
 
+// clang-format off
+__attribute__((deprecated))
 char* trimPrefix(char* str);
+// clang-format on
```

Comment:
While this works in this case, do we prefer "comment-based" deprecation?

Reply:
> What do you mean by "comment-based" deprecation?

Something like this: https://github.com/irods/irods/blob/cf4744ecd539f3ea3847f957d8d1121a468100d9/lib/core/include/irods/rcMisc.h#L350-L356

Reply:
I see. Check the deprecation rules.
- https://docs.irods.org/4.3.2/developers/deprecation/

I don't know if they cover this situation. Let me think about this and get back to you.

Reply:
Probably need to tweak the deprecation notes regarding always providing a message since deprecation implies avoiding usage (see https://github.com/irods/irods/pull/7811#discussion_r1645232619).

I'll write up an issue for that.

Reply:
Believe this is answered, leaving open just in case though.

Reply:
Looks like it has been addressed, yep.

Reply:
Resolving.